ZipBuilder/Sources/ZipBuilder/ZipBuilder.swift

```diff
@@ -128,42 +128,12 @@ struct ZipBuilder {
     self.customSpecRepos = customSpecRepos
   }
 
-  // TODO: This function contains a lot of "copy these paths to this directory, fail if there are
-  //   errors" code. It could probably be broken out into a cleaner interface or broken out into
-  //   separate functions.
-  /// Try to build and package the contents of the Zip file. This will throw an error as soon as it
-  /// encounters an error, or will quit due to a fatal error with the appropriate log.
-  ///
-  /// - Returns: Information related to the built artifacts.
-  /// - Throws: One of many errors that could have happened during the build phase.
-  func buildAndAssembleRelease(inProjectDir projectDir: URL) throws -> ReleaseArtifacts {
-    // If it exists, remove it before we re-create it. This is simpler than removing all objects.
-    if FileManager.default.directoryExists(at: projectDir) {
-      try FileManager.default.removeItem(at: projectDir)
-    }
-
-    CocoaPodUtils.podInstallPrepare(inProjectDir: projectDir)
-
-    // Get the README template ready (before attempting to build everything in case this fails,
-    // otherwise debugging it will take a long time).
-    let readmePath = paths.templateDir.appendingPathComponent(Constants.ProjectPath.readmeName)
-    let readmeTemplate: String
-    do {
-      readmeTemplate = try String(contentsOf: readmePath)
-    } catch {
-      fatalError("Could not get contents of the README template: \(error)")
-    }
-
+  func buildAndAssembleZip(podsToInstall: [CocoaPodUtils.VersionedPod]) ->
```

Comment:
Nit: Can you please add documentation to the func declaration?

Reply:
Done.